var/spack/repos/builtin.mock/packages/zlib-ng/package.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.build_systems import autotools, cmake
+from spack.package import *
+
+
+class ZlibNg(AutotoolsPackage, CMakePackage):
```

Comment:
Do we need something as complex as this to make the tests work? It seems like unnecessary overhead, and we don't want to introduce this as a zlib-api provider in the mock repo since so many tests hardcode zlib.

Reply:
No not really. I was going for simple and tried to grab a low level package. We need a git attribute, with git ref versions and variants. I originally wrote this with modifying some of the git mock packages but it ended up being invasive on a lot of tests so I just grabbed something else. I can rename it to some dummy package and pair it down.